docs/articles/documentation/why-testcafe/README.md

```diff
@@ -0,0 +1,126 @@
+
+### The Server-Client Architecture
+
+The server-client architecture of TestCafe allows it to execute system-level and in-browser code.
+
+* TestCafe uses [high-level system APIs](https://github.com/DevExpress/testcafe-browser-tools) to launch and manage browsers. This is necessary to achieve full control over test execution.
+* TestCafe tests are full-fledged Node.js scripts. They can launch services and applications, read and write system files, make use of your favourite libraries.
+* At the same time, TestCafe is capable of performing in-browser actions with client-side automation scripts. This is how TestCafe handles asynchronous events, simulates user activity, and executes custom (user-defined) JavaScript.
+
+Our experience shows that this hybrid approach gives users the best of both worlds. Early versions of TestCafe ran entirely in the browser. This caused a number of issues. The tests’ capabilities were limited, the application often interfered with page execution, and the sub-optimal logic separation caused browser crashes. Since then, we decoupled the engine, exposed the underlying Node.js API, and implemented a test runner that executes test scripts server-side.
+
+### Page Proxying: Bridging the Server-Client Gap
+
+You may notice that when you run TestCafe, the URL in the browser’s address bar does not match that of your website. This happens because TestCafe runs an under-the-hood reverse proxy.
+
+The [testcafe-hammerhead](https://github.com/DevExpress/testcafe-hammerhead) proxy intercepts browser requests and adds automation scripts to the requested resources. It then modifies the URLs contained within the resource, so that they point to the proxy. This way, neither the client-side code, nor the resources it communicates with, can tell that the page has been modified. To conceal automation scripts from the rest of the page code, TestCafe mocks the browser API.
+
+The proxying mechanism ensures that the page appears to be hosted at the original URL even to the test code. This is why you can use your website’s actual URL in tests, and pay no mind to the browser’s address bar.
+
+### Browser Sandboxing
+
+At the end of each run, TestCafe deletes all browser cookies, empties the storage, and reloads the page, thereby preventing undesirable interference with subsequent tests. You don’t need to write boilerplate code to reset the app state and reverse the changes your tests make.
+
+Tests that run in parallel operate in independent sandboxed environments. This helps prevent server-side collisions.
+
+### Client-Side Scripts
+
+To perform common testing tasks, TestCafe translates server-side calls to its API into client-side code. However, some scenarios require the execution of client-side code that TestCafe cannot automatically generate. That’s why the framework offers multiple ways to execute user-defined JavaScript.
```

Comment:
```suggestion
To perform basic testing tasks, TestCafe translates server-side API calls into client-side code. Some complex scenarios, however, require the execution user-defined JavaScript code. You can embed such code in a number of ways:
```

Reply:
a) Client functions aren't an *advanced* feature per se, and is appealing for users of all skill levels. Your test scenarios don't have to be that complex for you to try it in action.
b) We're describing several different kinds of code injections, none of which are represented particularly well by the word `embed` :(

Reply:
We can remove the word complex and change the last part: 

```suggestion
To perform basic testing tasks, TestCafe translates server-side API calls into client-side code. Some scenarios, however, require the execution user-defined JavaScript code. TestCafe allows you to run/inject such scripts in a number of ways:
```

Reply:
The first sentence describes the TestCafe API in its entirety - so it would be incorrect to label the tasks in question 'basic'. 
```suggestion
TestCafe translates server-side test code into client-side JavaScript, and injects it into the browsers that it controls. This process enables the framework to perform common in-browser actions. Some testing scenarios, however, require the execution of custom client-side code. There are three ways to do it with TestCafe:
```